### PR TITLE
[CLI] Update CLI binary doc site to include openssl3 instruction

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Added the instruction on how to download openssl3 manually. In case users see any errors regarding to openssl3 when running the CLI binary.

### CODE REVIEW

**Comment by banool** [comment]
No longer necessary once I land the PR to remove OpenSSL from the MacOS CLI.

**Comment by banool** [comment]
Though I suppose it can't hurt to have, let's land it for now.

**Review by banool — APPROVED** [review]

File: developer-docs-site/docs/tools/aptos-cli/install-cli/download-cli-binaries.md [hunk 1]
```diff
@@ -35,6 +35,27 @@ These instructions have been tested on macOS Monterey (12.6)
...
+[1]    81095 abort      aptos
+```
+
+Install `openssl3` with the following command:
+1. Download latest version from https://www.openssl.org/source/
+2. Unzip the downloaded file
+3. `cd` into the unzipped folder e.g. `cd openssl-3.1.2`
+4. Run `./config --prefix /usr/local darwin64-x86_64-cc` configure openssl3
+5. Run `make` to build openssl3
+6. Run `sudo make install` to install openssl3. Notice that `sudo` is required to install openssl3 to `/usr/local` folder.
```
Comment (saharct) [inline developer-docs-site/docs/tools/aptos-cli/install-cli/download-cli-binaries.md:54]:
```suggestion
Take the following steps to install `openssl3`:
1. Download the latest version from [OpenSSL](https://www.openssl.org/source).
2. Unzip the downloaded file.
3. `cd` into the unzipped folder, for example: `cd openssl-3.1.2`
4. Run `./config --prefix /usr/local darwin64-x86_64-cc` to configure openssl3.
5. Run `make` to build openssl3.
6. Run `sudo make install` to install openssl3. Notice that `sudo` is required to install openssl3 to the `/usr/local` folder.
```

File: developer-docs-site/docs/tools/aptos-cli/install-cli/download-cli-binaries.md [hunk 2]
```diff
@@ -35,6 +35,27 @@ These instructions have been tested on macOS Monterey (12.6)
 1. Follow the simple steps recommended by the Apple support in [Open a Mac app from an unidentified developer](https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/mac) to remove the "unknown developer" blocker.
 1. Type `~/bin/aptos help` to read help instructions.
 1. Add `~/bin` to your path in your `.bashrc` or `.zshrc` file for future use.
+1. Run `aptos help` to see the list of commands and verify that the CLI is working.
+
+Note: If you encounter an error message like the following. You will need to manually install `openssl3`.
```
Comment (saharct) [inline developer-docs-site/docs/tools/aptos-cli/install-cli/download-cli-binaries.md:40]:
```suggestion
Note: You will need to manually install `openssl3` if you encounter an error message like the following:
```

**Review by saharct — APPROVED** [review]

**Pull request merged**